pkg/blockbuilder/blockbuilder_metrics.go

```diff
@@ -22,6 +26,27 @@ func newBlockBuilderMetrics(reg prometheus.Registerer) blockBuilderMetrics {
 		NativeHistogramBucketFactor: 1.1,
 	})
 
+	m.processPartitionDuration = promauto.With(reg).NewHistogramVec(prometheus.HistogramOpts{
+		Name:                        "cortex_blockbuilder_process_partition_duration_seconds",
+		Help:                        "Time spent processing one partition.",
+		NativeHistogramBucketFactor: 1.1,
+	}, []string{"partition"})
+
+	m.compactAndUploadDuration = promauto.With(reg).NewHistogramVec(prometheus.HistogramOpts{
+		Name:                        "cortex_blockbuilder_compact_and_upload_duration_seconds",
+		Help:                        "Time spent compacting and uploading one partition.",
+		NativeHistogramBucketFactor: 1.1,
+	}, []string{"partition"})
+
+	m.fetchRecordsTotal = promauto.With(reg).NewCounter(prometheus.CounterOpts{
+		Name: "cortex_blockbuilder_fetch_records_total",
+		Help: "Total number of records received by the consumer.",
+	})
+	m.fetchErrors = promauto.With(reg).NewCounter(prometheus.CounterOpts{
+		Name: "cortex_blockbuilder_fetch_errors_total",
```

Comment:
Wouldn't it be confusing to run mimir in monolith mode, if both `ingester` and `blockbuilder` packages used the same metrics?

Reply:
idk, i don't think so. You'd have a single metric to look at for errors. Otherwise you always need to check two metrics.

It's also an option to unify the metrics when we end up refactoring the block builder to be able to consume faster with the replay speed work

Reply:
Yeah, let's come back to the metrics naming separately.